presto-main/src/main/java/com/facebook/presto/operator/scalar/ScalarFunctionImplementation.java

```diff
@@ -47,6 +55,14 @@ public ScalarFunctionImplementation(boolean nullable, List<Boolean> nullableArgu
             Class<?> instanceType = instanceFactory.get().type().returnType();
             checkArgument(instanceType.equals(methodHandle.type().parameterType(0)), "methodHandle is not an instance method");
         }
+
+        // check if nullableArguments and nullFlags match
+        checkArgument(nullFlags.size() == nullableArguments.size(), "the sizes of nullableArguments and nullFlags must match");
```

Comment:
The message is not very user-friendly. `nullableArguments` and `nullFlags` don't mean anything to the author of a function (assuming this is the exception that would show up if the function is implemented incorrectly)